### PR TITLE
feat: Detect the number of issues created that have a bounty

### LINKED ISSUE #15: Detect the number of issues created that have a bounty, show in column "Bountied Issue", add "Bountied Issue $" column
/bounty $20

### PR DESCRIPTION
**Pull request by Anshgrover23**
/claim #15
Fixes #15

### CODE REVIEW

File: lib/getBountiedIssues.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  repo: string,
+  contributor: string,
+  startDate: string
+): Promise<{ number: number; amount: number }[]> {
+  try {
+    const { data: issues } = await octokit.issues.listForRepo({
+      owner: repo.split('/')[0],
+      repo: repo.split('/')[1],
+      creator: contributor,
+      since: startDate,
```
Comment (seveibar) [inline lib/getBountiedIssues.ts:31]: Hmm why not filter by tag

Comment (seveibar) [reply lib/getBountiedIssues.ts:31]: I see you do it below, but wondering if its possible here to avoid pagination likits

**Review by seveibar — CHANGES_REQUESTED** [review]
Can you demonstrate that this works and remove the package lock (we use bun in all projects)

File: scripts/test-github-issues.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { Octokit } from "@octokit/rest"
+
+const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN })
+
+// Function to extract bounty amount from comment body
+function extractBountyAmountFromComment(
+  commentBody: string | undefined,
+): number {
```
Comment (seveibar) [inline scripts/test-github-issues.ts:8]: Import code under test, dont redefine it

Comment (seveibar) [reply scripts/test-github-issues.ts:8]: Goes for almost this entire file

**Review by seveibar — CHANGES_REQUESTED** [review]

File: lib/getBountiedIssues.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+function extractBountyAmount(issue: any): number {
+  const bodyMatch = issue.body?.match(/\$(\d+)\s*bounty/i)
+  if (bodyMatch) {
+    return parseInt(bodyMatch[1])
+  }
+  return 0 // Default if no bounty found
+}
+
+// Fetch bountied issues for a specific contributor
+export async function fetchBountiedIssues(
```
Comment (seveibar) [inline lib/getBountiedIssues.ts:16]: Function name should match file name

File: lib/getBountiedIssues.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        })
+
+        // Find the comment by 'algora-pbc[bot]' and extract the bounty amount
+        const botComment = comments.find((comment: any) => comment.user.login === "algora-pbc[bot]")
+        // console.log(botComment);
+        
+        
+        // Extract the bounty amount from the bot's comment
+        const amount = botComment && botComment.body ? extractBountyAmountFromComment(botComment.body) : 0
+        console.log(botComment?.body);
```
Comment (seveibar) [inline lib/getBountiedIssues.ts:59]:
```suggestion
```

**Comment by seveibar** [comment]
@Anshgrover23 can you show the result of actually generating the table? It's been broken in the past by people not fully testing, i'd like to make sure it actually generates the table properly.

**Comment by seveibar** [comment]
`bun scripts/generate-weekly-overview.ts` should do it.

Remember that when you PR code, it is expected that you've run the code, never PR code you haven't run.

I've actually only ever suspended one contributor from tscircuit, and it was for not running their code and repeatedly introducing bugs. So this is a very critical/important thing to do.

**Comment by imrishabh18** [comment]
@Anshgrover23 I don't think the data is correct, see the issues created column. It's 0 for all

**Review by seveibar — APPROVED** [review]

**Pull request merged**